tests/components/roon/test_config_flow.py

```diff
@@ -3,164 +3,157 @@
 
 from homeassistant import config_entries, setup
 from homeassistant.components.roon.const import DOMAIN
-from homeassistant.const import CONF_HOST
 
-from tests.common import MockConfigEntry
 
+class RoonHubMock:
+    """Class to mock the Roon API for testing."""
 
-class RoonApiMock:
-    """Mock to handle returning tokens for testing the RoonApi."""
-
-    def __init__(self, token):
+    def __init__(self, token, servers):
         """Initialize."""
         self._token = token
+        self._servers = servers
+
+    async def discover(self):
+        """Return discovered roon servers."""
+        return self._servers
+
+    async def authenticate(self, host, servers):
+        """Authenticate against roon."""
+        return (self._token, "core_id")
+
 
-    @property
-    def token(self):
-        """Return the auth token from the api."""
-        return self._token
-
-    def stop(self):  # pylint: disable=no-self-use
-        """Close down the api."""
-        return
-
-
-async def test_form_and_auth(hass):
-    """Test we get the form."""
-    await setup.async_setup_component(hass, "persistent_notification", {})
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
-    assert result["type"] == "form"
-    assert result["errors"] == {}
-
-    with patch("homeassistant.components.roon.config_flow.TIMEOUT", 0,), patch(
-        "homeassistant.components.roon.const.AUTHENTICATE_TIMEOUT",
-        0,
-    ), patch(
-        "homeassistant.components.roon.config_flow.RoonApi",
-        return_value=RoonApiMock("good_token"),
-    ), patch(
-        "homeassistant.components.roon.async_setup", return_value=True
-    ) as mock_setup, patch(
+class RoonHubMockException(RoonHubMock):
+    """Class to mock the Roon API for testing that throws an unexpected exception."""
+
+    async def authenticate(self, hass, host, servers):
+        """Throw exception when authenticating."""
+        raise Exception
+
+
+async def test_successful_discovery_and_auth(hass):
+    """Test when discovery and auth both work ok."""
+    with patch(
+        "homeassistant.components.roon.config_flow.RoonHub",
```

Comment:
Don't patch the code in the config flow. Patch the library. If we need to test timeouts it's ok to patch the timeout constant only.